Homework 4

Zarrass opened the pull request:
Ксожалению я так и не понял, как заставить обновляться секунды онлайн

Zarrass committed:
Homework 4

Ксожалению я так и не понял, как заставить обновляться секунды онлайн

Bael commented on sis-angular/src/app/first/first.component.html:
```diff
@@ -29,6 +29,8 @@ <h2>А теперь об интересах и  увлечениях:</h2>
 <h1>
-  Спасибо за внимание.
+  Спасибо за внимание.<br>
+  До нового года осалось:<br>
```
осталось

Bael commented on sis-angular/src/app/first/first.component.html:
```diff
@@ -29,6 +29,8 @@ <h2>А теперь об интересах и  увлечениях:</h2>
-  Спасибо за внимание.
+  Спасибо за внимание.<br>
+  До нового года осалось:<br>
+   {{monthLast}} месяцев, {{dayLast}} дней, {{hoursLast}} часов, {{minutesLast}} минут, {{secondsLast}} секунд.
```
чтобы упростить данный шаблон, можно использовать pipe Для часов минут и секунд, например: 
{{dateLast | date:'hh:mm:ss'}}

Bael commented on sis-angular/src/app/first/first.component.ts:
```diff
@@ -1,15 +1,42 @@
 })
 export class FirstComponent implements OnInit {
+  aboutme:AboutmeClass;
+  dateNY:number = new Date(2019, 0, 1).getTime();
```
данный код выполнится один раз, при создании компонента.

Bael commented on sis-angular/src/app/first/first.component.ts:
```diff
@@ -1,15 +1,42 @@
+  }
 
   ngOnInit() {
+    this.timerId = setInterval(() => this.dateResult--, 1000);
```
1) dateResult измеряется в миллисекундах, а вы отнимаете 1, получается что замедляете прогресс в тысячу раз :) 
2) толку от этого нет, потому что нет пересчета остатка по дням месяцам и минутам.

Bael reviewed (changes requested):
Михаил, вы близки к успешному завершению работы, надо немного поправить расчет остатка, основной совет: выделите расчет остатка в отдельный метод, вызывайте его в конструкторе и в интервальной функции.
Давайте доведем работу до конца :)

Zarrass committed:
Uodate homework4

Вот что у меня получилось с функцией, она все равно не обновляет онлайн и мне пока не понятно, почему(

Bael commented on sis-angular/src/app/first/aboutme-class.ts:
```diff
@@ -0,0 +1,13 @@
+export class AboutmeClass {
```
можно было убрать class из названия, просто AboutMe

Bael commented on sis-angular/src/app/first/first.component.html:
```diff
@@ -29,6 +29,9 @@ <h2>А теперь об интересах и  увлечениях:</h2>
+  Спасибо за внимание.<br>
+  До нового года осталось:<br>
+  {{dateLast | date:'h:mm:ss'}}
+   {{monthLast}} месяцев, {{dayLast}} дней, {{hoursLast}} часов, {{minutesLast}} минут, {{secondsLast}} секунд.
```
{{hoursLast}} часов, {{minutesLast}} минут, {{secondsLast}} 
теперь не нужны же - надо убрать, 
 {{dateLast | date:'h:mm:ss'}} вместо них.

Bael commented on sis-angular/src/app/first/first.component.ts:
```diff
@@ -1,15 +1,57 @@
+  }
 
   ngOnInit() {
+   this.timerId = setInterval(this.dateresult(this.dateNY,this.dateNow), 1000);
```
в setInterval надо передавать функцию, а не ее использование. Поэтому у вас и не работало. 
Т.е. надо this.timerId = setInterval(() => this.dateresult(this.dateNY,this.dateNow), 1000);

Bael commented on sis-angular/src/app/first/first.component.ts:
```diff
@@ -1,15 +1,57 @@
+  minutesLast:number;
+  secondsLast:number;
+
+    dateresult (dateNY:number,dateNow:number ): Date {
```
есть вопрос к названию, я бы просто назвал updateTimer и ничего бы не возвращал - просто обновлял бы dateLast и количество месяцев и дней.

Bael reviewed (changes requested):
Давайте доведем до конца:) Хотя бы прикрутите лямбду в setInterval.
В целом надо уделить внимание к наименованиям переменных и функций. 
Еще посоветую использовать автоформатирование в  Intelli Idea, будет проще работать намного :).

Zarrass committed:
Update2 Homework4

Таймер, именование, имплементация. Убрал пайп

Bael reviewed (approved):
ок!
